### DISCUSSION TITLE
Concatination of text to make URL link text

### DISCUSSION THREAD

**Discussion by ArthurMitchell42** [JanDeDobbeleer/oh-my-posh#4035] [Ideas]
Hi,
One idea I had was to create a URL link using string concatenation to open an SFTP session at the current path and machine, and then place that around the "runner@host" segment. In order to do this I think the following would be needed?

{{ url {{.UserName}}@{{.HostName}} \"sftp://\"+.UserName+\"@\"+.HostName+\"/\".Location }}

but I can't seem to see any way of concatenating strings to make this work?
Arthur

**Comment by JanDeDobbeleer** [comment]
Have a look at the [join function](http://masterminds.github.io/sprig/string_slice.html) in sprig.

**Reply by ArthurMitchell42** [reply]
Hi Jan, 
I figured it out and got this working:

"template": "{{ if not .Writable }} \uf456{{ end }} {{if .Segments.Session.SSHSession}}{{ $sftp := list \"sftp://\" .UserName \"@\" .HostName .Segments.Path.Location | join \"\"}}{{ url .Path $sftp }}{{else}}{{ path .Path .Location }}{{end}} ",

One conversion I can't get working is string length. When I use 
   {{$l:=len .HostName}} 
Then test it with
   {{if gt $l 8}}

It complains that $l is a sting, not integer. I tried go string to int conversion functions but they don't seem to work on the template line. Do you have any advice please?

Thanks, Arthur

**Reply by JanDeDobbeleer** [reply]
This works for me: `{{ $l:=len .HostName }}{{ if gt $l 5 }}yup {{ end }}`

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/2492783/5bd64477-6a80-4306-89ce-9bb24e1a3e55)
